### PR TITLE
mysql: remove unused options and reorder args

### PR DESCRIPTION
**Pull request by raymondelooff**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

- Remove the unused `--with-archive-storage-engine` and `--with-blackhole-storage-engine` options, as those storage engines are enabled by default
- Remove the `pidof` dependency, as MySQL only needs it on Linux. On Mac it uses `pgrep`.
- Reorder the args in an alphabetical order
- Remove the `mysql-tests` directory only when not built with `--with-tests`

I've tested the formula with both a clean installation and upgrade of an existing installation.

### CODE REVIEW

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

**Comment by DomT4** [comment]
The test changes here seem like a bit of a downgrade on the status quo? The latter actually exercises functionality, albeit _minimal functionality_ I admit, whereas the new test simply asks it to confirm the version. Is that not the case or?

**Comment by raymondelooff** [comment]
During the installation, the `mysql-test-run.pl` script is used to start the server, instead of the previous test script (that may be unreliable given the `sleep` method, that assumes the database will always start within 2 seconds). The MySQL test script starts the server and asks for the status. The `mariadb` and `percona-server` formulae use and can use the same test method, so it is consistent between all different MySQL forks.

When the `mysql-test` directory is removed during the installation (and that's by default, because it is a very large directory), the same method can not be used later on in the `test` method, so in that case it will only call the executable to check whether the correct version of MySQL was built. When building with `--with-test`, the `mysql-test` directory will not be removed. In that case, the test method uses the MySQL script to test the server.

**Comment by DomT4** [comment]
> During the installation, the mysql-test-run.pl script is used to start the server, instead of the previous test script 

I understand this, but people have made what I'd consider a valid point before that having a comprehensive test suite during install doesn't nullify the usefulness of having one inside `test do` as well, not least because the latter is flexible to Homebrew and can be modified locally & run repeatedly in quick succession without having to rebuild the whole formula.

> (that may be unreliable given the sleep method, that assumes the database will always start within 2 seconds)

Are there significant known cases where it _didn't_ start within 2 seconds? Just thinking about the amount of times that test has been through CI without issues to date.

> (and that's by default, because it is a very large directory)

Heh, run `git blame Formula/mysql.rb`, I'm familiar 😉 😄.

I think this PR is generally solid and certainly harmonisation across the different MySQL forks is useful, but yeah, I'm just not sure about the logic behind castrating the existing test block.

**Comment by raymondelooff** [comment]
Yeah, valid points. Thanks for the feedback! The castration idea came from the `mariadb` formula and seemed like a more reliable way to test the formula, but yeah, it doesn't really test anything useful when built without tests. I will revert the changes to the `test` method for this formula. I will add the same `test` method to the `percona-server` formula in PR #22098.

> Heh, run git blame Formula/mysql.rb, I'm familiar 😉 😄.

I see 😄

**Comment by DomT4** [comment]
I'm completely happy for @ilovezfs to overrule me here if he wants to, this is his yard these days; I just know I've used the `mysql` test block to poke around before locally and I'd be surprised if I was the only one. `mysql` is not a pleasant formula to rebuild from source particularly frequently 🙈.

The `mariadb` test is older than the `mysql` one, IIRC, so it could be the other way around, that `mariadb` could benefit from adopting this format.

File: Formula/mysql.rb [hunk 1]
```diff
@@ -22,19 +23,18 @@ class Mysql < Formula
...
-  option "with-blackhole-storage-engine", "Compile with the BLACKHOLE storage engine enabled"
   option "with-local-infile", "Build with local infile loading support"
-  option "with-debug", "Build with debug support"
+  option "with-memcached", "Build with InnoDB Memcached plugin"
+  option "with-test", "Build with unit tests"
 
-  deprecated_option "enable-local-infile" => "with-local-infile"
   deprecated_option "enable-debug" => "with-debug"
+  deprecated_option "enable-memcached" => "with-memcached"
+  deprecated_option "enable-local-infile" => "with-local-infile"
```
Comment (ilovezfs) [inline Formula/mysql.rb]: why is this second

Comment (raymondelooff) [reply Formula/mysql.rb]: Didn't see that one, thanks!

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

**Review by ilovezfs — APPROVED** [review]

**Comment by ilovezfs** [comment]
Thanks for the great PR @raymondelooff! Shipped.

**Pull request closed without merging**